### DISCUSSION TITLE
AttributeError: module 'tweepy' has no attribute 'StreamListener'

### DISCUSSION THREAD

**Discussion by taimurkhaneth** [tweepy/tweepy#2115] [Questions] [answered]
I have a twitter AI bot that uses AI to do tweets on automatically but I having problem with tweepy if someone can help me solve this issue 

# main.py
import tweepy
import openai
from config import (TWITTER_API_KEY, TWITTER_API_SECRET_KEY,
                    TWITTER_ACCESS_TOKEN, TWITTER_ACCESS_TOKEN_SECRET,
                    OPENAI_API_KEY)

class MentionStreamListener(tweepy.StreamListener):
    def __init__(self, api):
        self.api = api
        self.me = api.me()

    def on_status(self, tweet):
        print(f"{tweet.user.screen_name} mentioned you: {tweet.text}")

        if tweet.user.id == self.me.id:
            return

        prompt = f"Reply to the tweet: {tweet.text}"
        response = openai.Completion.create(
            engine="text-davinci-003",
            prompt=prompt,
            max_tokens=100,
            n=1,
            stop=None,
            temperature=0.7,
        )
        reply_content = response.choices[0].text.strip()
        reply = f"@{tweet.user.screen_name} {reply_content}"
        self.api.update_status(status=reply, in_reply_to_status_id=tweet.id)

        print(f"Replied to {tweet.user.screen_name}: {reply}")

# Set up Twitter API connection
auth = tweepy.OAuthHandler(TWITTER_API_KEY, TWITTER_API_SECRET_KEY)
auth.set_access_token(TWITTER_ACCESS_TOKEN, TWITTER_ACCESS_TOKEN_SECRET)
twitter_api = tweepy.API(auth)

# Set up OpenAI API connection
openai.api_key = OPENAI_API_KEY

# Monitor mentions and reply
mention_stream_listener = MentionStreamListener(twitter_api)
mention_stream = tweepy.Stream(twitter_api.auth, mention_stream_listener)
mention_stream.filter(track=[f"@{twitter_api.me().screen_name}"], is_async=True)

**Comment by Harmon758** [comment, marked as answer]
`StreamListener` was merged into `Stream` in [Tweepy v4.0.0](https://tweepy.readthedocs.io/en/v4.14.0/changelog.html#version-4-0-0-2021-09-25), almost two years ago.

https://tweepy.readthedocs.io/en/v4.14.0/streaming.html

For code block usage, see https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks.

**Comment by taimurkhaneth** [comment]
so If they changed it what can I do with my bot that it start working @Harmon758

**Comment by Harmon758** [comment]
[The Twitter API v1.1 statuses/filter endpoint has been deprecated](https://twittercommunity.com/t/announcing-the-deprecation-of-v1-1-statuses-filter-endpoint/182960), so if you want to use streaming, you'll have to use Twitter API v2.
